tutorials/flappy-bird/step_8/flappy-bird.cpp

```diff
@@ -0,0 +1,660 @@
+        if (player_died) return;
+
+        // Loop all columns to check collisions between player and the pipes
+        for (auto entity : registry.view<graphics::layer<3>>()) {
+            // Check collision between player and a collidable object
+            if (collisions::basic_collision_system::query_rect(registry, player, entity)) {
+                // Mark player died as true
+                player_died = true;
+            }
+        }
+    }
+
+private:
+    entt::entity player;
+    bool &player_died;
+};
+
+// Name this system
+REFL_AUTO (type(collision_logic));
+
+// Game Scene
+class game_scene final : public scenes::base_scene {
+public:
+    game_scene(entt::registry &registry, ecs::system_manager &system_manager_) noexcept : base_scene(registry),
+                                                                                          system_manager(
+                                                                                                  system_manager_) {
+        // Set the constants that will be used in the program
+        registry.set<flappy_bird_constants>();
+
+        // Create everything
+        score_entity = create_score(registry);
+        create_background(registry);
+        init_dynamic_objects(registry);
+    }
+
+    // Scene name
+    std::string scene_name() noexcept final {
+        return "game_scene";
+    }
+
+private:
+    // Update the game every tick
+    void update() noexcept final {
+        // Retrieve constants
+        const auto constants = entity_registry_.ctx<flappy_bird_constants>();
+
+        // Check if jump key is tapped
+        bool jump_key_tapped = input::virtual_input::is_tapped("jump");
+
+        // If game is not started yet and jump key is tapped
+        if (!started_playing && jump_key_tapped) {
+            // Game starts, player started playing
+            started_playing = true;
+            resume_physics();
+        }
+
+        // If player died, game over, and pause physics
+        if (player_died) {
+            player_died = false;
+            game_over = true;
+            pause_physics();
+        }
+
+        // If game is over, and jump key is pressed, reset game
+        if (game_over && jump_key_tapped) reset_game();
+    }
+
+    // Initialize dynamic objects, this function is called at start and resets
+    void init_dynamic_objects(entt::registry &registry) {
+        create_columns(registry);
+
+        // Create player
+        auto player = create_player(registry);
+
+        // Create systems
+        system_manager.create_system<column_logic>(score_entity);
+        system_manager.create_system<player_logic>(player);
+        system_manager.create_system<collision_logic>(player, player_died);
+
+        // Disable physics and everything at start to pause the game
+        pause_physics();
+
+        // Reset state values
+        started_playing = false;
+        player_died = false;
+        game_over = false;
```

Comment:
Here there is 4 step

- Creating player (1 line of code we keep it)
- Creating systems (separate in a functions)
- Disabling Physics (1line of code we keep it)
- Resetting values (Separate in a function)
Initiating a player, creating systems, and reset values